src/service/vaults.ts

```diff
@@ -372,3 +382,71 @@ export const makeOpenVaultOffer = async (
     toast.error('Unable to propose offer.');
   }
 };
+
+type Proposal = {
+  give: {
+    Collateral?: { amount: CapData<string> };
+    Minted?: { amount: CapData<string> };
+  };
+  want: {
+    Collateral?: { amount: CapData<string> };
+    Minted?: { amount: CapData<string> };
+  };
+};
+
+type AdjustParams = {
+  vaultOfferId: string;
+  collateral?: { amount: Amount<'nat'>; action: CollateralAction };
+  debt?: { amount: Amount<'nat'>; action: DebtAction };
+};
+
+export const makeAdjustVaultOffer = async ({
+  vaultOfferId,
+  collateral,
+  debt,
+}: AdjustParams) => {
+  const invitationSpec = {
+    source: 'continuing',
+    previousOffer: vaultOfferId,
+    invitationMakerName: 'AdjustBalances',
+  };
+
+  const { importContext, offerSigner } = appStore.getState();
+
+  const proposal: Proposal = { give: {}, want: {} };
```

Comment:
This is not contract code but it's still good practice to make the proposal deeply immutable.
So after the mutations call `harden(proposal)` to convey that it isn't to change anymore.

Reply:
Done